### PR TITLE
DuckDB-Wasm fixes related to (auto)loadable extensions

### PR DESCRIPTION
**Pull request by carlopi**
None of those changes should be introduce any functional change when DUCKDB_WASM is not defined.

Missing is a review of error messages around (auto)loading, since it's misleading to instruct users to perform `INSTALL`. Will do that in a separate & independent PR.

### CODE REVIEW

File: src/function/table/system/test_all_types.cpp [hunk 1]
```diff
@@ -204,13 +204,14 @@ vector<TestType> TestAllTypesFun::GetTestTypes(bool use_large_enum) {
 	auto max_map_value = Value::MAP(ListType::GetChildType(map_type), map_values);
 	result.emplace_back(map_type, "map", std::move(min_map_value), std::move(max_map_value));
 
+#ifndef DUCKDB_WASM
```
Comment (Mytherin) [inline src/function/table/system/test_all_types.cpp:207]: Is this necessary? Can't we add the `EXCLUDE` on the DuckDB-WASM side?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - one comment:

File: src/function/table/system/test_all_types.cpp (hunk 1 above)
Comment (carlopi) [reply src/function/table/system/test_all_types.cpp:207]: Makes sense, will do that.

**Comment by carlopi** [comment]
Thanks for the review and comment! For me this is ready.

**Pull request merged**